common/flatpak-dir.c

```diff
@@ -2078,6 +2078,8 @@ flatpak_dir_system_helper_call_deploy (FlatpakDir         *self,
 {
   const char *empty[] = { NULL };
 
+  if (arg_subpaths == NULL)
```

Comment:
I think we similarly need this `NULL` checking in `upgrade_deploy_data()` for the `subpaths` there?

Reply:
Never mind, `flatpak_deploy_data_get_subpaths()` is guaranteed to return something `non-NULL`